boilerplate/services/notification.service.js

```diff
@@ -1,13 +1,24 @@
+const { OBJECT_TYPES } = require('@semapps/activitypub');
 const urlJoin = require('url-join');
 const { SingleMailNotificationsService } = require('@semapps/notifications');
 const QueueService = require('moleculer-bull');
 const CONFIG = require('../config/config');
 const transport = require('../config/transport');
+const notificationFilter = require('./mixins/MailNotificationFilterMixin');
 
+/**
+ * Service for sending mails.
+ * Its `SingleMailNotificationsService` mixin listens to inbox events
+ * which triggers mails to recipients.
+ * If there is a mapper from the activity-mapping service, and if it
+ * is not filtered by `filterNotification`.
+ * The `notificationFilter` mixin filters activity notifications
+ * from ignored contacts.
+ */
 module.exports = {
   mixins: CONFIG.QUEUE_SERVICE_URL
-    ? [SingleMailNotificationsService, QueueService(CONFIG.QUEUE_SERVICE_URL)]
-    : [SingleMailNotificationsService],
+    ? [SingleMailNotificationsService, notificationFilter, QueueService(CONFIG.QUEUE_SERVICE_URL)]
+    : [SingleMailNotificationsService, notificationFilter],
```

Comment:
Unfortunately there is still a problem. In Moleculer the order of the mixins is important. The notificationFilter mixin must be before the SingleMailNotificationsService, otherwise the `filterNotification` method of SingleMailNotificationsService overwrites the `filterNotification` method of the notificationFilter mixin. This is a bit contrary to common sense !